src/translationedit.cpp

```diff
@@ -82,36 +82,38 @@ bool TranslationEdit::parseTranslationData(QOnlineTranslator *translator)
                     wordLine.append(QStringLiteral(": <font color=\"grey\"><i>%1</i></font>").arg(translations.join(QStringLiteral(", "))));
 
                 // Add generated line to edit
-                append(wordLine);
+                formattedTranslation += wordLine;
             }
 
             indent.setTextIndent(0);
             textCursor().setBlockFormat(indent);
-            append({}); // Add a new line before the next type of speech
+            formattedTranslation += "<br>"; // Add a new line before the next type of speech
         }
     }
 
     // Examples
     if (!translator->examples().isEmpty()) {
-        append(QStringLiteral("<font color=\"grey\"><i>%1</i> - %2</font>").arg(translator->source(), tr("examples:")));
+        formattedTranslation += QStringLiteral("<font color=\"grey\"><i>%1</i> - %2</font>").arg(translator->source(), tr("examples:"));
         const QMap<QString, QVector<QExample>> examples = translator->examples();
         for (auto it = examples.cbegin(); it != examples.cend(); ++it) {
-            append(QStringLiteral("<b>%1</b>").arg(it.key()));
+            formattedTranslation += QStringLiteral("<b>%1</b>").arg(it.key());
             QTextBlockFormat indent;
             indent.setTextIndent(20);
             textCursor().setBlockFormat(indent);
             for (const auto &[example, description] : it.value()) {
-                append(description);
-                append(QStringLiteral("<font color=\"grey\"><i>%1</i></font>").arg(example));
-                append({});
+                formattedTranslation += description;
+                formattedTranslation += QStringLiteral("<font color=\"grey\"><i>%1</i></font>").arg(example);
+                formattedTranslation += "<br>";
             }
             indent.setTextIndent(0);
             textCursor().setBlockFormat(indent);
         }
     }
 
+    setHtml(formattedTranslation);
+
```

Comment:
Nitpick: could you please remove this line to alight setHtml with `moveCursor`?
```suggestion
```

Reply:
Done.